src/vs/workbench/services/editor/common/editorOpenWith.ts

```diff
@@ -155,7 +155,21 @@ export function getAllAvailableEditors(
 
 					const fileEditorInput = editorService.createEditorInput({ resource, forceFile: true });
 					const textOptions: IEditorOptions | ITextEditorOptions = options ? { ...options, override: false } : { override: false };
-					return { override: editorService.openEditor(fileEditorInput, textOptions, group) };
+					return {
+						override: (async () => {
+							// Try to replace existing editors for resource
+							const existingEditor = group.editors.find(editor => editor.resource && isEqual(editor.resource, resource));
+							if (existingEditor && !fileEditorInput.matches(existingEditor)) {
+								await editorService.replaceEditors([{
+									editor: existingEditor,
+									replacement: fileEditorInput,
+									options: options ? EditorOptions.create(options) : undefined,
+								}], group);
+							}
+
+							return editorService.openEditor(fileEditorInput, textOptions, group);
```

Comment:
I believe this extra `openEditor` call would not be needed since the `replaceEditors` call should take care of that already, please test.

Reply:
This function currently needs to return an `IEditorPane`, so I was calling `openEditor` to get that. I also seem to remember that replaceEditor does not show the replacing editor, although I could be wrong